### DISCUSSION TITLE
Plugin Development - addFieldNamePicker should return both the data frame name and field name

### DISCUSSION THREAD

**Discussion by 6nv** [grafana/grafana#63541] [Plugins]
The strings that `addFieldNamePicker` currently returns are ambiguous. Please see this screenshot for a summary of my confusion, using the Static datasource for demonstration purposes. It is unclear which names correspond to which data series, so I strongly believe the field name picker should return a structure containing both the data frame name **and** field name instead of a string.
<img width="1913" alt="grafana-confusion" src="https://user-images.githubusercontent.com/91281131/220491535-b14aa03e-fbcb-415c-9d93-531e45add64d.png">

**Comment by 6nv** [comment]
Referenced in [this Grafana forum thread](https://community.grafana.com/t/clarity-on-fieldnamepicker/91502).

Turns out, sometimes FieldNamePicker returns the value of `frame.config.displayNameFromDS` as well:

<img width="1470" alt="more-field-name-confusion" src="https://github.com/grafana/grafana/assets/91281131/3d17f42b-765e-42ad-9017-b575df8f19c2">

Ideally, the field name in this example would be "rh".

**Comment by 6nv** [comment]
Bump

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). We are migrating this discussion to an issue and closing the discussion. The issue is #82581. Feel free to continue the discussion around this there. Thank you!
